packages/gatsby/cache-dir/loading-indicator/indicator.js

```diff
@@ -0,0 +1,64 @@
+import React from "react"
+import Portal from "./portal"
+import Style from "./style"
+import { isLoadingIndicatorEnabled } from "$virtual/loading-indicator"
+import { debugLog } from "../debug-log"
+
+if (typeof window === `undefined`) {
+  throw new Error(
+    `Loading indicator should never be imported in code that doesn't target only browsers`
+  )
+}
+
+if (module.hot) {
+  module.hot.accept(`$virtual/loading-indicator`, () => {
+    // isLoadingIndicatorEnabled is imported with ES import so no need
+    // for dedicated handling as HMR just replace it in that case
+  })
+}
+
+// HMR can rerun this, so check if it was set before
+// we also set it on window and not just in module scope because of HMR resetting
+// module scope
+if (typeof window.___gatsbyDidShowLoadingIndicatorBefore === `undefined`) {
+  window.___gatsbyDidShowLoadingIndicatorBefore = false
+}
+
+export function Indicator({ visible = true }) {
+  if (!isLoadingIndicatorEnabled()) {
+    return null
+  }
+
+  if (!window.___gatsbyDidShowLoadingIndicatorBefore) {
+    // not ideal to this in render function, but that's just console info
+    debugLog(
+      `You might have seen "gatsby develop" loading indicator in the bottom left corner.\n\nIf you want to disable it you can visit ${window.location.origin}/___loading-indicator/disable (will disable it for current session).`
```

Comment:
Slight edit to the message: 
      `A loading indicator is displayed in-browser whenever content is being requested upon navigation (Query On Demand). You can disable the loading indicator for your current session by visiting ${window.location.origin}/___loading-indicator/disable`

Reply:
@pieh - what do you think of the minor edit? Do you think it misses any detail, or creates confusion?

Reply:
We do show loading indicator only if request doesn't finish within 1000 miliseconds - this is some "perceived performance" trick that I've read about somewhere that showing indicator immediately make it feels as less performant than it actually is (or maybe rather delaying showing indicator make it feel more performant) - I don't know if it's important to capture it in the message. So we show it when request is taking longer than what could be described as "almost instant" (maybe 1000 miliseconds is to high threshold for that, but we already do have precedent in using this value for some APIs that allow users to integrate their own custom loaders)

Reply:
But overall - I think this is improvement over my message, so I will push update soon ;) And we can always iterate more if needed